src/datasources/transaction-api/transaction-api.service.spec.ts

```diff
@@ -325,6 +333,128 @@ describe('TransactionApi', () => {
+      const safe = safeBuilder().build();
+      const cacheDir = new CacheDir(
+        `${chainId}_safe_exists_${safe.address}`,
+        '',
+      );
+      const isSafe = faker.datatype.boolean();
+      cacheService.get.mockResolvedValueOnce(JSON.stringify(isSafe));
+      networkService.get.mockResolvedValueOnce({ status: 200, data: safe });
+
+      const actual = await service.isSafe(safe.address);
+
+      expect(actual).toBe(isSafe);
+      expect(cacheService.get).toHaveBeenCalledTimes(1);
+      expect(cacheService.get).toHaveBeenCalledWith(cacheDir);
+      expect(networkService.get).not.toHaveBeenCalled();
+      expect(cacheService.set).not.toHaveBeenCalledTimes(1);
+    });
+
+    it('should return false if Safe does not exist', async () => {
+      const safe = safeBuilder().build();
+      const cacheDir = new CacheDir(
+        `${chainId}_safe_exists_${safe.address}`,
+        '',
+      );
+      cacheService.get.mockResolvedValueOnce(undefined);
+      networkService.get.mockResolvedValueOnce({ status: 404, data: null });
+
+      const actual = await service.isSafe(safe.address);
+
+      expect(actual).toBe(false);
+      expect(cacheService.get).toHaveBeenCalledTimes(1);
+      expect(cacheService.get).toHaveBeenCalledWith(cacheDir);
+      expect(networkService.get).toHaveBeenCalledTimes(1);
+      expect(networkService.get).toHaveBeenCalledWith({
+        url: `${baseUrl}/api/v1/safes/${safe.address}`,
+      });
+      expect(cacheService.set).toHaveBeenCalledTimes(1);
+      expect(cacheService.set).toHaveBeenCalledWith(
+        cacheDir,
+        'false',
+        indefiniteExpirationTime,
+      );
+    });
+
+    const errorMessage = faker.word.words();
+    it.each([
+      ['Transaction Service', { nonFieldErrors: [errorMessage] }],
+      ['standard', new Error(errorMessage)],
+    ])(`should forward a %s error`, async (_, error) => {
+      const safe = safeBuilder().build();
+      const getSafeUrl = `${baseUrl}/api/v1/safes/${safe.address}`;
+      const statusCode = faker.internet.httpStatusCode({
+        types: ['serverError'],
+      });
+      const expected = new DataSourceError(errorMessage, statusCode);
+      const cacheDir = new CacheDir(
+        `${chainId}_safe_exists_${safe.address}`,
+        '',
+      );
+      cacheService.get.mockResolvedValueOnce(undefined);
+      networkService.get.mockRejectedValueOnce(
+        new NetworkResponseError(
+          new URL(getSafeUrl),
+          {
+            status: statusCode,
+          } as Response,
+          error,
+        ),
+      );
+
+      await expect(service.isSafe(safe.address)).rejects.toThrow(expected);
+
+      expect(cacheService.get).toHaveBeenCalledTimes(1);
+      expect(cacheService.get).toHaveBeenCalledWith(cacheDir);
+      expect(networkService.get).toHaveBeenCalledTimes(1);
+      expect(networkService.get).toHaveBeenCalledWith({
+        url: `${baseUrl}/api/v1/safes/${safe.address}`,
+      });
```

Comment:
Nit: I'd also test here that `cacheService.set` is not called when an error in the upstream service occurs, wdyt?

Reply:
Added in 43fb39a6923480a82d15146cb4317eb806821f1f.